core/src/main/java/org/apache/carbondata/scan/result/vector/CarbonColumnVector.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.carbondata.scan.result.vector;
+
+import org.apache.spark.sql.types.Decimal;
+
+public interface CarbonColumnVector {
```

Comment:
Why is it an interface? Are you considering make it offheap in the future? If not, I think use class directly is enough